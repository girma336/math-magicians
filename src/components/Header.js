import React from 'react';
import { Link } from 'react-router-dom';
import './styles/Header.css';

const Header = () => (
  <div>
    <header className="header-logo">
      <div className="logo">
        Math Magicians
      </div>
      <nav>
        <ul>
          <li><Link to="/">Home</Link></li>
          <li><Link to="/calculator">Calculator</Link></li>
          <li><Link to="/quote">Quote</Link></li>
        </ul>
      </nav>
    </header>
  </div>
);

export default Header;
