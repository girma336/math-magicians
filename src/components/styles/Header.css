header {
  width: 100%;
  height: 60px;
  display: flex;
  position: fixed;
  align-items: center;
  justify-content: space-between;
  top: 0;
  left: 0;
  box-shadow: 0 0 5px #666;
}

.logo {
  margin-left: 5%;
  font-size: 20px;
}

header nav {
  margin-right: 20px;
}

.header-logo ul {
  display: flex;
  justify-content: space-around;
  list-style: none;
}

header nav li {
  margin-right: 10px;
}
