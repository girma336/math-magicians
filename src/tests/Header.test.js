import React from "react";
import renderer from 'react-test-renderer';
import Header from "../components/Header";
import { BrowserRouter } from "react-router-dom";
describe("Test snapshot component of Header", () => {
    it("renders correctly of Header", () => {
        const tree = renderer.create(
        <BrowserRouter>
        <Header />
        </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})