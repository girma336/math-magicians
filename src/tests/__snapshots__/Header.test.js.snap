// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot component of Header renders correctly of Header 1`] = `
<div>
  <header
    className="header-logo"
  >
    <div
      className="logo"
    >
      Math Magicians
    </div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
